.github/workflows/cran-status-check.yaml

```diff
@@ -0,0 +1,65 @@
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 6 * * 1,3,5' # Runs at 06:00 on Mon/Wed/Fri
+
+name: check-cran-status
+
+jobs:
+  fetch-deadlines:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    permissions:
+      issues: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          packages: gh
+
+      - name: Check for existing CRAN issues
+        id: check-issues
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # Count open issues with CRAN-related labels
+          ISSUE_COUNT=$(gh issue list --label "cran-deadline" --state open --json number | jq length)
+          if [ $ISSUE_COUNT -eq 0 ]; then
+            SHOULD_RUN="true"
+            echo "✅ Will run CRAN check"
+          else
+            SHOULD_RUN="false"
+            echo "⏭️  Skipping CRAN check - existing issues found"
+          fi
+          echo "should-run=$SHOULD_RUN" >> $GITHUB_OUTPUT
+
+      - name: Fetch deadline for this package
+        if: steps.check-issues.outputs.should-run == 'true'
+        shell: Rscript {0}
+        run: |
+          crandb <- tools::CRAN_package_db()
+
+          pkgname <- drop(read.dcf("DESCRIPTION", "Package"))
+
+          deadline <- crandb[crandb$Package == pkgname, "Deadline"]
+
+          if (!is.na(deadline)) {
+            gh::gh(
+              "POST /repos/{owner_repo}/issues",
+              owner_repo = Sys.getenv("GITHUB_REPOSITORY"),
+              title = paste("Fix CRAN R CMD check issues by", deadline),
+              body = paste0(
```

Comment:
maybe link to the GHA logs if possible? or at least annotate that this GHA is responsible for the issue.

That will help debug if this should generate a false positive or get some metadata wrong, etc.

Reply:
Fixed by [`02ade03` (#7175)](https://github.com/Rdatatable/data.table/pull/7175/commits/02ade03ca38df653216fa2afc810c216e05678fe)